airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_tracker/TrackingMetadata.java

```diff
@@ -102,9 +111,50 @@
             metadata.put("volume_rows", syncSummary.getRecordsSynced());
           }
         }
+
+        final List<FailureReason> failureReasons = failureReasonsList(attempts);
+        if (!failureReasons.isEmpty()) {
+          metadata.put("failure_reasons", failureReasonsListAsJson(failureReasons).toString());
+          metadata.put("main_failure_reason", failureReasonAsJson(failureReasons.get(0)).toString());
+        }
       }
     }
     return metadata.build();
   }
 
+  private static List<FailureReason> failureReasonsList(final List<Attempt> attempts) {
+    return attempts
+        .stream()
+        .map(Attempt::getFailureSummary)
+        .flatMap(Optional::stream)
+        .map(AttemptFailureSummary::getFailures)
+        .flatMap(Collection::stream)
+        .toList();
+  }
+
+  private static ArrayNode failureReasonsListAsJson(final List<FailureReason> failureReasons) {
+    return Jsons.arrayNode().addAll(failureReasons
+        .stream()
+        .sorted(Comparator.comparing(FailureReason::getTimestamp))
```

Comment:
I think this sort should happen in `failureReasonsList()` instead of here, because right now I don't think the `main_failure_reason` is actually using this sorted list at all!

Reply:
oh oops, my bad!